docs/data/tree-view/simple-tree-view/customization/CustomContentTreeView.js

```diff
@@ -1,84 +1,70 @@
-import clsx from 'clsx';
 import { styled } from '@mui/material/styles';
-import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import Avatar from '@mui/material/Avatar';
 import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
-import { TreeItem, useTreeItemState } from '@mui/x-tree-view/TreeItem';
+import { useTreeItem } from '@mui/x-tree-view/internals/useTreeItem';
+import {
+  TreeItemNextContent,
+  TreeItemNextIconContainer,
+  TreeItemNextTransitionGroup,
+  TreeItemNextLabel,
+  TreeItemNextRoot,
+} from '@mui/x-tree-view/internals/TreeItemNext';
+import { TreeItemIcon } from '@mui/x-tree-view/internals/TreeItemIcon';
+import { TreeItemProvider } from '@mui/x-tree-view/internals/TreeItemProvider';
 
-const CustomContentRoot = styled('div')(({ theme }) => ({
-  '&': { padding: theme.spacing(0.5, 1) },
+const CustomTreeItemContent = styled(TreeItemNextContent)(({ theme }) => ({
+  padding: theme.spacing(0.5, 1),
 }));
 
-const CustomContent = React.forwardRef(function CustomContent(props, ref) {
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
   const {
-    className,
-    classes,
-    label,
+    id,
     nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
+    label,
+    children,
+    slots = {},
+    slotProps = {},
+    ...other
   } = props;
 
   const {
-    disabled,
-    expanded,
-    selected,
-    focused,
-    handleExpansion,
-    handleSelection,
-    preventSelection,
-  } = useTreeItemState(nodeId);
-
-  const icon = iconProp || expansionIcon || displayIcon;
-
-  const handleMouseDown = (event) => {
-    preventSelection(event);
-  };
-
-  const handleClick = (event) => {
-    handleExpansion(event);
-    handleSelection(event);
-  };
+    getRootProps,
+    getContentProps,
+    getIconContainerProps,
+    getLabelProps,
+    getGroupTransitionProps,
+    status,
+  } = useTreeItem({ id, nodeId, children, label, rootRef: ref });
 
   return (
-    <CustomContentRoot
-      className={clsx(className, classes.root, {
-        'Mui-expanded': expanded,
-        'Mui-selected': selected,
-        'Mui-focused': focused,
-        'Mui-disabled': disabled,
-      })}
-      onClick={handleClick}
-      onMouseDown={handleMouseDown}
-      ref={ref}
-    >
-      <div className={classes.iconContainer}>{icon}</div>
-      <Box sx={{ flexGrow: 1, display: 'flex', gap: 1 }}>
-        <Avatar
-          sx={(theme) => ({
-            background: theme.palette.primary.main,
-            width: 24,
-            height: 24,
-            fontSize: '0.8rem',
-          })}
-        >
-          {label[0]}
-        </Avatar>
-        <Typography component="div" className={classes.label}>
-          {label}
-        </Typography>
-      </Box>
-    </CustomContentRoot>
+    <TreeItemProvider nodeId={nodeId}>
+      <TreeItemNextRoot {...getRootProps(other)}>
+        <CustomTreeItemContent {...getContentProps()} ownerState={status}>
```

Comment:
I followed the Base UI approach, so the `ownerState` is not part of the `getContentProps`, which make sense when you use a custom component for the DOM structure because it will probably not want this `ownerState`.
But if you use the built-in component or one derived from it (like `CustomTreeItemContent` here), then you do need to pass the `ownerState` in order to have the variants.

And it gets even worse if we require the `ownerState` on all the other slots (which we probably should if we want to properly support `styleOverrides`.

@michaldudak what would be your take on this one?
I feel like our needs are not exactly the same as the one from a Base UI users, most people will do things like in this demo, i.e: use `useTreeItem` to play Tetris with the blocks of the `TreeItem`, maybe remove one, replace one but most likely not all of them.

Reply:
By the way, the same applies with classes, in this demo, the `CustomTreeItemContent` never receives the classes like `treeItemClasses.selected` because it's apply in the `useSlotProps`.
So I moved all the styling logic to `variants` to not use those classes for our own styling, which means that this demo work and people can manually pass the classes if **they** need them.

Reply:
I'm wondering why do you need both a component (TreeItemNextContent) and a hook. Doesn't the component handle setting its own props?

Reply:
`useTreeItem` is the hook to build you own `TreeItemNext`, not `TreeItemNextContent`.

We are trying to provide a good DX for people who need to create custom versions of the `TreeItemNext` component.
The idea here is that slots are often not enough, they don't allow you to easily swap two elements or to insert an element between two others ([this doc example](https://deploy-preview-11721--material-ui-x.netlify.app/x/react-tree-view/simple-tree-view/customization/#headless-api)).
So we are providing a `useTreeItem` hook that allows you to create your own component.

We are trying to make the barrier to entry low for this new hook, because IMHO it will be useful for simple customization (the demo linked above is a good example), and not just for people create a whole new design system.
So being able to easily re-use the blocks used by `TreeItemNext` (`TreeItemNextContent`, `TreeItemNextLabel`, etc...) would be quite important.

Reply:
I'm wondering how the DX should look like for the use-cases.
I'm trying to create something consistent with Base as much as possible to avoid ending with yet another DX for customization.

Reply:
An alternative would be that `TreeViewNextContent` gets the `status` as a prop and uses it instead of the `ownerState` for its styling, with a `shouldForwardProp` to avoid passing it to the DOM.

The main downside here is that people creating a custom component that receives `getContentProps` should make sure not to forward to the DOM.
But it would only impact advanced customizations

Reply:
https://github.com/mui/mui-x/pull/11721/commits/caeeb8abe22506c283360e7a141f1cbfcfbd2a7d

@michaldudak do you think it's a good approach?

Reply:
> I'm trying to create something consistent with Base as much as possible to avoid ending with yet another DX for customization.

With the component-per-node approach, Base UI won't use `ownerState` as currently. It'll be only used as a callback to the render prop if you want to override the component being rendered (it won't be passed to subcomponents by their parents).

> https://github.com/mui/mui-x/commit/caeeb8abe22506c283360e7a141f1cbfcfbd2a7d
> @michaldudak do you think it's a good approach?

It does look better to me.